fix excluded tests

this fails: `pytest -k slow`

```
FAILED tests/test_sweeps.py::test_hydra_sweep_ddp_sim
FAILED tests/test_sweeps.py::test_optuna_sweep_ddp_sim_wandb - Failed: [I 2023-02-14 14:05:17,404] A new study created in memory with name: no-name-f7c4a0c9-3dd3-487a-8a13-0c2c19d9c4c8
FAILED tests/test_train.py::test_train_ddp_sim - torch.multiprocessing.spawn.ProcessRaisedException: 
=============================================== 3 failed, 14 passed, 4 skipped in 1166.59s (0:19:26) ================================================

```

--- a/src/datamodules/datamodule.py
+++ b/src/datamodules/datamodule.py
@@ -75,9 +75,16 @@ class PieDataModule(LightningDataModule, Generic[DocumentType, InputEncoding, Ta
             raise TypeError("IterableTaskEncodingDataset has no length")
         return len(data_train)
 
-    def prepare_data(self):
+    def setup(self, stage: str):
 
-        for split in [self.train_split, self.val_split, self.test_split]:
+        if stage == "fit":
+            split_names = [self.train_split, self.val_split]
+        elif stage == "test":
+            split_names = [self.test_split]
+        else:
+            raise NotImplementedError(f"not implemented for stage={stage} ")
+
+        for split in split_names:
             if split is None or split not in self.dataset:
                 continue
             task_encoding_dataset = self.taskmodule.encode(

